sdjournal/journal.go

```diff
@@ -260,6 +262,61 @@ func (j *Journal) GetDataValue(field string) (string, error) {
 	return strings.SplitN(val, "=", 2)[1], nil
 }
 
+// GetDataMap returns all the Key-Value pairs of data for the current journal
+// entry. Mimics the behavior of output_export in logs-show.c
+func (j *Journal) GetDataMap() (map[string]string, error) {
+	var r C.int
+	var d unsafe.Pointer
+	var l C.size_t
+
+	resultMap := make(map[string]string)
+
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	var realtimeUsec C.uint64_t
```

Comment:
The idea was to replicate the behavior of the journalctl export/json format messages. This would be a valid object if json serialized to be dumped into another journal. Although, the structure t implementation would allow keeping this behavior by default (and .String() could implement the export format)